extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js

```diff
@@ -223,14 +227,31 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
   }
 
   /**
-   * Force this extension to behave as if ad is about to be shown.
-   * Used for testing but should be extended to force an ad to show in a real story.
-   * @param {string} pageBeforeAdId
+   * Force this extension to render all amp-story-auto-ads controlled UI and
+   * navigate to ad.
+   * @param {string=} pageBeforeAdId
    * @visibleForTesting
    */
   forceRender(pageBeforeAdId) {
+    const pageBeforeId =
+      pageBeforeAdId ||
+      /** @type {string} */ (this.storeService_.get(
+        StateProperty.CURRENT_PAGE_ID
+      ));
     this.isCurrentAdLoaded_ = true;
-    this.tryToPlaceAdAfterPage_(pageBeforeAdId);
+    // Setting distance manually to avoid flash of next page.
+    const lastPage = this.adPageEls_[this.adPageEls_.length - 1];
```

Comment:
worth to extract a method and shared in array.js

`function lastItem(array)`

Reply:
Done.